table/server/common/src/main/java/alluxio/table/common/transform/action/WriteAction.java

```diff
@@ -69,8 +75,26 @@ private WriteAction(String type, int numFiles) {
 
   @Override
   public JobConfig generateJobConfig(Layout base, Layout transformed) {
-    // TODO: set partition info
-    return new CompactConfig(null, base.getLocation().toString(),
+    PartitionInfo partitionInfo;
+    try {
+      partitionInfo = ProtoUtils.toHiveLayout(base.toProto());
+    } catch (InvalidProtocolBufferException e) {
+      return new CompactConfig(null, base.getLocation().toString(),
+          transformed.getLocation().toString(),
+          mLayoutType, mNumFiles);
+    }
+    String serdeClass = partitionInfo.getStorage().getStorageFormat().getSerde();
+    String inputFormat = partitionInfo.getStorage().getStorageFormat().getInputFormat();
+
+    ArrayList<SchemaField> colList = new ArrayList<>(partitionInfo.getDataColsList().size());
+    for (FieldSchema col : partitionInfo.getDataColsList()) {
+      colList.add(new SchemaField(col.getId(), col.getName(), col.getType(),
+          col.getOptional(), col.getComment()));
+    }
+    alluxio.job.transform.PartitionInfo transformPartInfo
+        = new alluxio.job.transform.PartitionInfo(serdeClass, inputFormat,
+        partitionInfo.getStorage().getSerdeParametersMap(), colList);
```

Comment:
does `getSerdeParametersMap` contain `"skip.header.line.count"="1"`

Reply:
it is a table property

Reply:
I verified in a local test, it's not contained in the map